libraries/node/consensus/include/final_chain/data.hpp

```diff
@@ -36,10 +37,22 @@ struct BlockHeader {
 
   HAS_RLP_FIELDS
 
-  static h256 const& uncles_hash();
-  static Nonce const& nonce();
-  static u256 const& difficulty();
-  static h256 const& mix_hash();
+  static h256 const& uncles_hash() { return EmptyRLPListSHA3(); }
+
+  static Nonce const& nonce() {
+    static const Nonce ret;
```

Comment:
correct me if I am wrong but this will created separate static variable for each file where it is included, so I find this absolutely in a conflict why was it coded like that. It needs to go to cpp file

Reply:
Do you guys understand what are these 3 static methods even good for because I dont ? :D Seems like it is supposed to return the same default value every time. In such case I would maybe change naming to default_nonce or something like that ?

Reply:
Can't remember why they were moved to the header file. They are needed for Ethereum compatibility if I understand it correctly. 
I moved this methods to cpp file and implement it the way uncles_hash method implemented